test/Unit/DependencyManagement/DependencyManagerTests.cs

```diff
@@ -229,40 +229,49 @@ public void StartDependencyInstallationIfNeeded_InstallsSnapshotInForeground_Whe
         [Fact]
         public void StartDependencyInstallationIfNeeded_InvokesBackgroundMaintainer_WhenAcceptableDependenciesAlreadyInstalled()
         {
-            _mockInstalledDependenciesLocator.Setup(_ => _.GetPathWithAcceptableDependencyVersionsInstalled())
-                .Returns("AlreadyInstalled");
-            _mockStorage.Setup(_ => _.GetDependencies()).Returns(GetAnyNonEmptyDependencyManifestEntries());
+            try
+            {
+                Environment.SetEnvironmentVariable("MDEnableAutomaticUpgrades", "true");
```

Comment:
If we do just this, we will not have a test that verifies this behavior when the `MDEnableAutomaticUpgrades` variable is _not_ present (which is actually the most common case).

I don't have a simple solution, let's think about it. Ideally, we need to test all combinations of `MDEnableAutomaticUpgrades` (true, false, absent), both before the EOL moment and after the EOL moment, which makes it 6 test cases. But then we'll have to pass both `MDEnableAutomaticUpgrades` and a fixed "current time" to the tests. Which is doable, but perhaps there is an easier way.

Reply:
I think there are more tests that are needed here, but wanted to get your thoughts on which ones are relevant. The test I added confirms that, if the Func<bool> passed to BackgroundDependencySnapshotMaintainer returns false, it does not attempt to install dependencies and instead warns to the logger. Any other cases you'd like me to add specific checks for?

Reply:
Thank you, I think the tests we have now are adequate.